### PR TITLE
reset functionality for stateful position limit handles

### PR DESCRIPTION
**Pull request by mathias-luedtke**
As discussed in #193 I have added reset functions to the stateful position limit handles.
Existing API was not touch, I have just added new methods to the handles and the corresponding interfaces.

### CODE REVIEW

File: joint_limits_interface/include/joint_limits_interface/joint_limits_interface.h [hunk 1]
```diff
@@ -554,10 +568,38 @@ class JointLimitsInterface : public hardware_interface::ResourceManager<HandleTy
...
 /** Interface for enforcing limits on a position-controlled joint with soft position limits. */
-class PositionJointSoftLimitsInterface : public JointLimitsInterface<PositionJointSoftLimitsHandle> {};
+class PositionJointSoftLimitsInterface : public JointLimitsInterface<PositionJointSoftLimitsHandle> {
+public:
+  /** \name Real-Time Safe Functions
+   *\{*/
+  /** \brief Reset all managed handles. */
+  void reset()
+  {
+    typedef hardware_interface::ResourceManager<PositionJointSoftLimitsHandle>::ResourceMap::iterator ItratorType;
```
Comment (adolfo-rt) [inline joint_limits_interface/include/joint_limits_interface/joint_limits_interface.h:595]: I think strict compilers will complain of a missing `typename` here. It should read `typedef typename hardware_interface...`

The above statement designates a nested type of a templated class, and compilers need help in figuring out that we're actually talking about a type here.

**Comment by adolfo-rt** [comment]
PR looks good except for one minor comment.

Could you also please extend the test suite to exercise the `reset()` methods, both for handles and interfaces?.

**Comment by mathias-luedtke** [comment]
I have added tests for bothinterfaces. The corresponding handles are tested implicitly.
Actually clang++complains if I add "typename", it is no dependent type name anymore.

**Comment by adolfo-rt** [comment]
I hope to give your tests a spin later today and propose a modification. As they stand, the new tests are basically checking that `reset()` compiles, but are not really checking that it does what it's supposed to. The changes to the test should be minor in implementation effort, but important in terms of validating what they should.

**Comment by adolfo-rt** [comment]
I take my last comment back. The tests are fine.

Thanks so much for the contribution with tests!

**Pull request merged**